lib/screen/workout/workout_list.dart

```diff
@@ -0,0 +1,198 @@
+        .map((workoutID) => workoutMap[workoutID])
+        .where((workout) => workout != null);
+    final userMadeWorkouts = workoutList.where((workout) => workout!.userMade);
+    final preMadeWorkouts = workoutList.where((workout) => !workout!.userMade);
+
+    final appBarActions = <Widget>[];
+
+    if (widget.enableModifying && !editMode) {
+      appBarActions.add(
+        IconButton(
+          onPressed: () => setState(() => editMode = true),
+          icon: const Icon(Icons.edit),
+        ),
+      );
+    }
+
+    return Scaffold(
+      resizeToAvoidBottomInset: false,
+      appBar: AppBar(
+        title: const Text('Workouts'),
+        actions: appBarActions,
+        leading: editMode
+            ? IconButton(
+                onPressed: () => setState(() => editMode = false),
+                icon: const Icon(Icons.check),
+              )
+            : null,
+      ),
+      body: CustomScrollView(
+        slivers: [
+          SliverPersistentHeader(floating: true, delegate: WorkoutSearchBar()),
+          SliverList(
+            delegate: SliverChildListDelegate([
+              WorkoutCategoryGroup(
+                action: widget.action,
+                category: 'Your Workouts',
+                workouts: List.from(userMadeWorkouts),
+                editMode: editMode,
+              ),
+              WorkoutCategoryGroup(
+                action: widget.action,
+                category: 'Try These Workouts',
+                workouts: List.from(preMadeWorkouts),
+                editMode: editMode,
+              ),
+            ]),
+          )
+        ],
+      ),
+      floatingActionButton: (widget.enableAdding && bloc.searchTerm == null)
+          ? FloatingActionButton(
+              onPressed: () => bloc.addNewWorkout(),
+              child: const Icon(Icons.add),
+            )
+          : null,
+    );
+  }
+}
+
+class WorkoutCategoryGroup extends StatefulWidget {
+  final void Function(Workout)? action;
+  final String category;
+  final List<Workout> workouts;
+  final bool editMode;
+
+  const WorkoutCategoryGroup({
+    Key? key,
+    required this.action,
+    required this.category,
+    required this.workouts,
+    required this.editMode,
+  }) : super(key: key);
+
+  @override
+  _WorkoutCategoryGroupState createState() => _WorkoutCategoryGroupState();
+}
+
+class _WorkoutCategoryGroupState extends State<WorkoutCategoryGroup>
+    with SingleTickerProviderStateMixin {
+  late final animation = AnimationController(
+    vsync: this,
+    duration: const Duration(milliseconds: 100),
+    value: 1,
+  );
+
+  late bool isExpanded = true;
+
+  @override
+  void dispose() {
+    animation.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    if ((isExpanded && animation.value != 1) ||
+        (!isExpanded && animation.value != 0)) {
+      animation.animateTo(isExpanded ? 1 : 0);
+    }
+
+    return AnimatedBuilder(
```

Comment:
Don't see the point of this AnimatedBuilder. AnimatedBuilder is to force a redraw everytime animation changes. Since you don't use the value of animation elsewhere, and just pass the object around, it's unnecessary.